zipkin-autoconfigure/ui/src/main/java/zipkin/autoconfigure/ui/ZipkinUiAutoConfiguration.java

```diff
@@ -75,6 +79,20 @@
   @Value("classpath:zipkin-ui/index.html")
   Resource indexHtml;
 
+  @Bean
+  String processedIndexHtml() throws IOException {
+    InputStream is = indexHtml.getInputStream();
```

Comment:
Should this be in a try block to ensure the stream is always closed even if there’s a parsing exception?

Reply:
In terms of Java best practices, yes, for sure. In terms of failing gracefully, I'm not sure what the expected behavior is for Spring applications: if we fail to parse the HTML, that most likely means the HTML is invalid. Should we then:

 * Blow up, fail early and fail hard?
 * Serve the HTML anyway, without modification (which we _know_ will fail behind a reverse proxy with the `base-path` config set?)
 * Serve the HTML anyway, but only if `base-path` is not set. If it's set, blow up?
 * Serve a custom error page that describes what happened, maybe even includes the exception?